Samples/Server/Unity-FractalStream/Assets/Scripts/WebRTCServer.cs

```diff
@@ -0,0 +1,179 @@
+        /// </summary>
+        public void Close()
+        {
+            if (!isClosing && plugin != null)
+            {
+                plugin.Dispose();
+                plugin = null;
+            }
+        }
+
+        /// <summary>
+        /// Handles input data and updates local transformations
+        /// </summary>
+        /// <param name="data">input data</param>
+        private void OnInputData(string data)
+        {
+            try
+            {
+                var node = SimpleJSON.JSON.Parse(data);
+                string messageType = node["type"];
+
+                switch (messageType)
+                {
+                    case "keyboard-event":
+                        var kbBody = node["body"];
+                        int kbMsg = kbBody["msg"];
+                        int kbWParam = kbBody["wParam"];
+
+                        break;
+
+                    case "mouse-event":
+                        var mouseBody = node["body"];
+                        int mouseMsg = mouseBody["msg"];
+                        int mouseWParam = mouseBody["wParam"];
+                        int mouseLParam = mouseBody["lParam"];
+
+                        break;
+
+                    case "camera-transform-lookat":
+                        string cam = node["body"];
+                        if (cam != null && cam.Length > 0)
+                        {
+                            string[] sp = cam.Split(new char[] { ' ', ',' }, StringSplitOptions.RemoveEmptyEntries);
+
+                            Vector3 loc = new Vector3();
+                            loc.x = float.Parse(sp[0]);
+                            loc.y = float.Parse(sp[1]);
+                            loc.z = float.Parse(sp[2]);
+                            location = loc;
+
+                            Vector3 look = new Vector3();
+                            look.x = float.Parse(sp[3]);
+                            look.y = float.Parse(sp[4]);
+                            look.z = float.Parse(sp[5]);
+                            lookAt = look;
+
+                            Vector3 u = new Vector3();
+                            u.x = float.Parse(sp[6]);
+                            u.y = float.Parse(sp[7]);
+                            u.z = float.Parse(sp[8]);
+                            up = u;
+                        }
+
+                        break;
+                }
+            }
+            catch (Exception ex)
+            {
+                Debug.LogWarning("InputUpdate threw " + ex.Message + "\r\n" + ex.StackTrace);
+            }
+        }
+
+        /// <summary>
+        /// Handles log data and writes it to the debug console
+        /// </summary>
+        /// <param name="logLevel">severity level</param>
+        /// <param name="logData">log data</param>
+        private void OnLogData(int logLevel, string logData)
+        {
+            Debug.Log("[" + logLevel + "]: " + logData);

```

Comment:
I suppose that we still need to add "Debug.isDebugBuild" condition so that it doesn't output in anything in Release build.

Reply:
we can't use `Debug.isDebugBuild` outside the main unity thread (so we can't use it here).

i'm not sure I agree in keeping this only in `debug` - there's value in logging this stuff in release too, right? My opinion is that this code is at the whim of the developer consuming our unity server script, so i think i'll abstract it into a separate optional behavior and then leave it up to the engineer

Reply:
Yes there is but it's unlikely right? There was actually some performance impact in the Unity HoloLens client when I tested last time if we keeps outputting the log every frame. However, this is not a big deal now since we've removed almost logs. Let's just skip it for now.